aspnetcore/web-api/index.md

```diff
@@ -453,7 +510,43 @@ To make automatic and custom responses consistent, call the <xref:Microsoft.AspN
 
 ### Log automatic 400 responses
 
-See [How to log automatic 400 responses on model validation errors (dotnet/AspNetCore.Docs#12157)](https://github.com/dotnet/AspNetCore.Docs/issues/12157).
+You can add logging by utilizing [InvalidModelStateResponseFactory](https://docs.microsoft.com/dotnet/api/microsoft.aspnetcore.mvc.apibehavioroptions.invalidmodelstateresponsefactory). Here is an example of code to be used in `Startup.ConfigureServices` that adds logging on top of **any** already existing functionality that generates the response:
+
+```csharp
+services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_2_2);
+
+services.PostConfigure<ApiBehaviorOptions>(options =>
+{
+    // If preserving of the existing functionality is desired, get a reference to the delegate.
+    var builtInFactory = options.InvalidModelStateResponseFactory;
+
+    options.InvalidModelStateResponseFactory = context =>
+    {
+        // As an example, we will create a logger with the full name of the action method as the category.
+        var logger = GetLoggerInstance(context.HttpContext.RequestServices, context.ActionDescriptor.DisplayName);
+
+        // Log accordingly.
+
+        // Use existing functionality - either one defined by us, or in this case the framework's.
+        return builtInFactory(context);
```

Comment:
Should we try and expand this code sample to also show how one can edit the returned object? It seems on ASP Net core 2.2 the object returned is [SerializableError](https://docs.microsoft.com/dotnet/api/microsoft.aspnetcore.mvc.serializableerror)

Reply:
As 2.2 is [no longer supported](https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core), we don't need to make changes to this version in the article. You can just revert this version of the section entirely, keeping only the first version that's for 3.0+.

Reply:
https://github.com/MohammadAsadi1372/MohammadAsadi.MasterService